src/charm_state.py

```diff
@@ -886,8 +887,14 @@ def check_use_aproxy(cls, use_aproxy: bool, values: dict) -> bool:
         Returns:
             Validated use_aproxy value.
         """
-        if use_aproxy and not (values.get("http") or values.get("https")):
-            raise ValueError("aproxy requires http or https to be set")
+        try:
+            if use_aproxy:
+                values_data = values.data  # type: ignore
```

Comment:
Here as well? Any linting disable rules should have an explanation why it is enabled (should be stated in our contributing guide)